OligoDesigner.R

```diff
@@ -0,0 +1,80 @@
+#Assign a 15 nucleotide target sequence in lowercase atcg format, using spaces between nucleotides
+target <- "c g g c g a c g t g a a c g g"
+
+#convert the target to vector and check for validity
+check_target <- function(target) {
+  target_v <- unlist(strsplit(target, split = " "))
+  print(target_v)
+  if (target_v[9] != "t") {
+    print("Error: the OMegazyme target must contain a U for RNA cleavage")
+  } else if (target_v[8] == "c") {
+    print("Warning! OMegazyme target should contain a A/G-U")
+  } else if (target_v[8] == "t") {
+    print("Warning! OMegazyme target should contain a A/G-U")
+  } else if (length(target_v) != 15) {
+    print("Error: the OMegazyme target must be 15 nucleotides in length")
+  } else {
+    print("The OMegazyme target is valid.")
+    return(target_v)
+  }
+}
+valid_target_v <- check_target(target)
+
+#reverse complement the target sequence
+rev_vtv <- rev(valid_target_v)
+complement_a <- function(x) {
+  if (x == "a") {
+    sub("a", "U", x)
+  } else {
+    return(x)
+  }}
+complement_t <- function(x) {
+  if (x == "t") {
+    sub("t", "A", x)
+  } else {
+    return(x)
+  }}
+complement_c <- function(x) {
+  if (x == "c") {
+    sub("c", "G", x)
+  } else {
+    return(x)
+  }}
+complement_g <- function(x) {
+  if (x == "g") {
+    sub("g", "C", x)
+  } else {
+    return(x)
+  }}
+rca <- lapply(rev_vtv, complement_a)
+rct <- lapply(rca, complement_t)
+rcc <- lapply(rct, complement_c)
+rc_target <- sapply(rcc, complement_g)
+rc_target
+
```

Comment:
Your code is easy to follow in this stepwise format, but something that is more concerted would be more effective if you were instead looking at many nucleotide target sequences.